src/helpers/date.js

```diff
@@ -19,3 +30,47 @@ export const formatTime = (time, format = "h:mm a") => {
 export const dateFromObjectId = (id) => {
     return dayjs(new Date(Number.parseInt(id.slice(0, 8), 16) * 1000));
 };
+
+export const now = (date, unit) => {
+    if (!date) {
+        if (unit) {
+            return dayjs.tz().startOf(unit);
+        }
+
+        return dayjs.tz();
+    }
+
+    if (typeof date === "number") {
+        const timestamp = date <= 2_147_483_647 ? date * 1000 : date;
+        return dayjs.tz(timestamp);
+    }
+
+    if (typeof date === "string") {
+        return dayjs.tz(date);
+    }
+
+    if (isDayjs(date)) {
+        // We do this late because under some conditions this is expensive (see: X2-9122)
+        return date;
+    }
+
+    if (date instanceof Date && !Number.isNaN(date.getTime())) {
+        return dayjs.tz(dateToString(date));
+    }
+
+    return dayjs.tz();
+};
+
+const padNumber = (value) => value.toString().padStart(2, "0");
+
+export const dateToString = (date) => {
+    const dateString = `${date.getFullYear()}-${padNumber(date.getMonth() + 1)}-${padNumber(date.getDate())}`;
+    const timeString = `${padNumber(date.getHours())}:${padNumber(date.getMinutes())}:${padNumber(date.getSeconds())}`;
+    return `${dateString} ${timeString}`;
+};
+
+export const getJSDate = (daysDate, isStartDate = true) => {
+    const time = isStartDate ? DayTimeStart : DayTimeEnd;
+
+    return new Date(formatDateWithTimezone(daysDate) + time);
```

Comment:
Can we do this pattern for EOD:
```js
dayjsDate.hour(0).minute(0).seconds(0).toDate();
dayjsDate.hour(23).minute(59).seconds(59).toDate();
```

Reply:
![image](https://github.com/xola/ui-kit/assets/24450223/14a6064e-acb7-49b4-9b0d-4317d3c9dbc4)
this adds the timezone offset when using dayjs.tz(). I checked others too and saw this one to be reliable.